packages/webapp/src/routes/developer/upload-metric-file.js

```diff
@@ -27,6 +30,17 @@ async function processSuccessfulUpload (result, request, h) {
   request.yar.set(constants.redisKeys.DEVELOPER_METRIC_DATA, result[0].metricData)
   request.yar.set(constants.redisKeys.DEVELOPER_METRIC_FILE_NAME, result.filename)
   logger.log(`${new Date().toUTCString()} Received metric data for ${result[0].location.substring(result[0].location.lastIndexOf('/') + 1)}`)
+  if (result[0].metricData?.d1 && filterByBGN(result[0].metricData?.d1, request).length === 0 && result[0].metricData?.e1 && filterByBGN(result[0].metricData?.e1, request).length === 0) {
```

Comment:
Changing the extra checks to `Array.isArray(result[0].metricData?.d1)` and `Array.isArray(result[0].metricData?.e1)` is better as it means `filterByBGN` will always return an Array so it will always have a `.length`